### ISSUE TITLE
Front end templates for Vet Center shows both common and official names

### ISSUE THREAD

**Issue by xiongjaneg** [department-of-veterans-affairs/va.gov-cms#15424]
## Problem statement
Veterans may know and search for their closest Vet Center by a common name, rather than an official name. Vet Center common names should be displayed along with official names to help Veterans identify and locate their Vet Centers.

## Background
The Outstation associated with Salt Lake City Vet Center is displayed using only the Official name. The Common name concept needs to be available for Outstations as well. 

Example: Major Brent Taylor Vet Center Outstation (node/17527), associated with [Salt Lake City Vet Center](https://www.va.gov/salt-lake-city-vet-center/locations/)

[Vet Center Sketch file](https://www.sketch.com/s/83e13c18-5a9f-40ed-860f-61b9ead58bd6)
[VBA Sketch file](https://sketch.com/s/891d33ae-152d-471c-ab5d-9aedf89cf6ff)

<details>
<summary> CMS and FE screenshots </summary>

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBZzNkIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--28c78c896ee1f339cda8dec32d062c36e2494f64/image.png)

Test cases should come from Tugboat data

</details>

## Current behavior 
- A Vet Center is identified only by the official name in the main location listing on it's own Locations page. 
   - Example 1: Dr. Cameron McKinley Department of Veterans Affairs Veterans Center (Prescott Vet Center) as displayed at the top of its own [Locations page](https://www.va.gov/prescott-vet-center/locations/)
   - Example 2: Col. Gail S. Halvorsen 'Candy Bomber' Veterans Center (Provo Vet Center) as displayed at the top of its own [Locations page](https://www.va.gov/provo-vet-center/locations/)
- ~A Vet Center is identified only by the official name when listed as a nearby Location on another Vet Center's page.~
   - ~Example 1: Dr. Cameron McKinley Department of Veterans Affairs Veterans Center (Prescott Vet Center) as displayed as a nearby location for [West Valley Vet Center](https://www.va.gov/west-valley-vet-center/locations/)~
   - ~Example 2: Col. Gail S. Halvorsen 'Candy Bomber' Veterans Center (Provo Vet Center) as displayed as a nearby location for [Salt Lake City Vet Center](https://www.va.gov/salt-lake-city-vet-center/locations/)~
   - The above behavior will be addressed in [this ticket.](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16309)
- An Outstation has only an official name 
  - Example: Major Brent Taylor Vet Center Outstation on [Salt Lake City Vet Center locations page](https://www.va.gov/salt-lake-city-vet-center/locations/) 
  
## Expected behavior
- Displayed names for Vet Centers and Outstations use both a official/legislated name and a common name.

## Acceptance Criteria

**Vet Center main page**
- [x] The Vet Center common name is used for the H1 title on the main Vet Center page
  - [ ]  If the common name and official name differ, show the "also called" statement with the official/legislative name
  - [ ] The common name should not be a semantic heading, it should use a `<p>` tag.
  - [ ] This official name must have an ID.
  - [ ] The H1 common name must have an `aria-describedby` attribute that references the ID of the official name element.
  - [ ] The `aria-describedby` attribute must be conditional - if there is no official name, the attribute should not be present
- [x] Design Review using demo environment https://web-txcrk8bym4a0ivw3xfr8xvdhezmls36f.demo.cms.va.gov/prescott-vet-center/
- [x] Accessibility Review using demo environment https://web-txcrk8bym4a0ivw3xfr8xvdhezmls36f.demo.cms.va.gov/prescott-vet-center/

**Vet Center Locations page**

- [x] Listings for a location with both an official and common name displays both names when listed as the Main location area on its own Locations page
- [x] Listings for a location with both an official and common name displays both names when listed as a Satellite locations on the main Vet  Center's Locations page
- ~Listings for a location with both an official and common name displays both names when listed as a nearby location on another Vet Center's Locations page~
- [x] All markup identified above (regarding the `id` and `aria-describedby` attributes) should be followed wherever the official name appears
- [x] Design review using demo environment https://web-txcrk8bym4a0ivw3xfr8xvdhezmls36f.demo.cms.va.gov/prescott-vet-center/locations/
- [x] Accessibility Review using demo environment https://web-txcrk8bym4a0ivw3xfr8xvdhezmls36f.demo.cms.va.gov/prescott-vet-center/locations/

## Team
Please check the team(s) that will do this work.

- [ ] CMS Team
- [ ] Public Websites
- [x] Facilities
- [ ] User support
- [ ] Accelerated Publishing

**Comment by mmiddaugh** [comment]
This is likely blocked until Outstations are able to have both an official and common name - will be resolved by [#15422 Vet Center Outstations should display common and official names](https://app.zenhub.com/workspaces/sitewide-facilities-639f5253e4b702a32376339e/issues/gh/department-of-veterans-affairs/va.gov-cms/15422)

**Comment by xiongjaneg** [comment]
@mmiddaugh Currently, both names (when present) are all read by a screen reader as H1, do we want this?

**Comment by laflannery** [comment]
@mmiddaugh To clarify the question here - The H1 on Vet Center pages has an `aria-describedby` attribute that points to the text below:
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/d90b5f5b-fed3-4349-88eb-5fa4e3b14def)

What this does is appends this text to the H1 when a screen reader user chooses to read it out. It will NOT make this the H1, so if you are looking at, or hearing, a list of Headings on the page you will still have only "Prescott Vet Center" but if the user reads through the page, this additional text will be associated with the H1:
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/aede12ae-d474-4b32-bded-1d30d857cbc3)

Maybe this doesn't clarify, but I tried lol

**Comment by laflannery** [comment]
I spoke with Michelle and she said:

> "known as name" is the legislated official name which we are obligated to use. We retained the "common" name for familiarity. Given this, I think I'm leaning toward announcing both. 

With that said, the markup should look as it currently does on [Prescott Vet Center](https://www.va.gov/prescott-vet-center/)
- The H1 has an `aria-describedby` that points to the ID of the `<p>` tag containing the Official name.
- However, this attribute should NOT show if there is no official name text. This is currently happening (i.e. [Salt Lake City Vet Center](https://www.va.gov/salt-lake-city-vet-center/))

**Comment by eselkin** [comment]
I have a few questions about common and official names...

In the example of the [Prescott Vet Center main page](https://www.va.gov/prescott-vet-center/)
We see the common name followed by Also known as the official name. With the AKA info attributing to the H1. This is correct. 

But on the [locations page](https://www.va.gov/prescott-vet-center/locations/) in the Main location section we only see the common name in the link. **Are we asking to also include the AKA official name in that link?** We currently have a very long if condition checking whether to include that as a URL or title.
```
if vetCenter.entityBundle == "vet_center_cap" or vetCenter.entityBundle == "vet_center" or vetCenter.entityBundle == "vet_center_outstation" or vetCenter.entityBundle == "vet_center_mobile_vet_center"
```
otherwise it displays it as a link as is being done in Prescott. But how would we want to display it here? I don't have examples of how that should look.

For the Other nearby Vet Centers do they need to display in this format?
Title
Also called Official name
Does it need to have aria-described by attribute on the Also called similar to the H1 in the main page? These are not H1s here, but H3s in vets-website. (Also checking if we have official names in that request that returns that data for the nearby list, because they are derived from a different source - vets-api).

**Comment by xiongjaneg** [comment]
blocked by #15660, which will likely be worked on in sprint 97

**Comment by mmiddaugh** [comment]
Now unblocked

**Comment by laflannery** [comment]
@xiongjaneg I updated the ACs with the required markup for this work. I know it's mentioned in a few other places (sketch, the comment above) but I thought it might be best to have it in the ACs to make sure it wasn't missed.

**Comment by maxx1128** [comment]
I'm putting some notes here about how I'm going to start addressing the code parts of this ticket:

1. Check the specific parts of `src/site/stages/build/drupal/graphql/vetCenterLocations.graphql.js` to make sure that `title` and `fieldOfficialName` are called on all of the needed queries. One of these is `reverseFieldOfficeNode` on line 87.
2. The markup on `src/site/includes/vet_centers/address_phone_image.liquid` needs to be updated to include the alternative name if it's present and different from the main name.

**Comment by jilladams** [comment]
Moving to Review / QA based on Max's scrum update today.

**Comment by maxx1128** [comment]
**Update:** In _current behavior,_ there are three examples of behavior to be changed. But example 2, showing both names for nearby locations is rendered through a React component on `vets-website`, and it also does its own API call through `vets-api` for the data. At [a lighthouse sync meeting](https://dsva.slack.com/archives/C0FQSS30V/p1701732517683809?thread_ts=1701711749.217649&cid=C0FQSS30V), it was decided that these changes are out of scope and more of a want than a need at the moment. So[ this pull request for the ticket](https://github.com/department-of-veterans-affairs/content-build/pull/1815) should be reviewed only for fixing examples 1 and 3 of the current behavior to the new behavior.

**Comment by jilladams** [comment]
@maxx1128 will you ticket or have you already tagged Jane to ticket the remaining work required for example 2 / React component?

**Comment by maxx1128** [comment]
I can write the ticket for the remaining work for this, and will reference this ticket too 👍

**Comment by maxx1128** [comment]
I created [the ticket for the remaining work](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16309) here and added an estimate while I was at it. cc: @xiongjaneg in case she wants to tweak any of the language or details.

**Comment by jilladams** [comment]
End of sprint update: Satellite offices AC is handled just we need sample content to prove it. Possibly mergeable today.

**Comment by jilladams** [comment]
PR is merged, should go out in today's daily content-build deploy

**Comment by xiongjaneg** [comment]
Design and a11y reviewed, all recommendations made and noted in PR, which was merged successfully
